src/main/java/FilterInput.java

```diff
@@ -0,0 +1,109 @@
+            break;
+        case Command.COMMAND_DONE:
+            int taskNumber;
+            checkDescription(words[FIRST_ARRAY_PARAMETER], descriptionInput);
+            taskNumber = Integer.parseInt(words[SECOND_ARRAY_PARAMETER]);
+            //might need to catch errors in the future
+            Greet.checkDoneTask(taskNumber);
+            break;
+        case Command.COMMAND_DELETE:
+            checkDescription(words[FIRST_ARRAY_PARAMETER], descriptionInput);
+            taskNumber = Integer.parseInt(words[SECOND_ARRAY_PARAMETER]);
+            Greet.deleteTask(taskNumber);
+            break;
+        case Command.COMMAND_TODO:
+            checkDescription(words[FIRST_ARRAY_PARAMETER], descriptionInput);
+            Todo todo = new Todo(descriptionInput[FIRST_ARRAY_PARAMETER]);
+            Greet.addTask(todo);
+            break;
+        case Command.COMMAND_DEADLINE:
+            checkDescription(words[FIRST_ARRAY_PARAMETER], descriptionInput);
+            checkTimeframe(descriptionInput);
+            Deadline deadline = new Deadline(descriptionInput[FIRST_ARRAY_PARAMETER],
+                    descriptionInput[SECOND_ARRAY_PARAMETER]);
+            Greet.addTask(deadline);
+            break;
+        case Command.COMMAND_EVENT:
+            checkDescription(words[FIRST_ARRAY_PARAMETER], descriptionInput);
+            checkTimeframe(descriptionInput);
+            Event event = new Event(descriptionInput[FIRST_ARRAY_PARAMETER],
+                    descriptionInput[SECOND_ARRAY_PARAMETER]);
+            Greet.addTask(event);
+            break;
+        default:
+            throw new DukeException(ErrorMessage.EXCEPTION_MESSAGE_UNKNOWN_COMMAND);
+        }
+    }
+
+    private static void checkDescription(String command, String[] descriptionInput) throws DukeException {
+
+        if (descriptionInput[FIRST_ARRAY_PARAMETER].equals("") ||
+                descriptionInput[FIRST_ARRAY_PARAMETER].equals(" ")) {
+            switch (command) {
+            case Command.COMMAND_TODO:
+                throw new DukeException(ErrorMessage.EXCEPTION_MISSING_DESCRIPTION_TODO);
+            case Command.COMMAND_DEADLINE:
+                throw new DukeException(ErrorMessage.EXCEPTION_MISSING_DESCRIPTION_DEADLINE);
+            case Command.COMMAND_EVENT:
+                throw new DukeException(ErrorMessage.EXCEPTION_MISSING_DESCRIPTION_EVENT);
+            default:
+                throw new DukeException(ErrorMessage.EXCEPTION_MESSAGE_MISSING_PARAMETERS_AFTER_COMMAND);
+            }
+        }
+    }
+
+    private static void checkTimeframe(String[] descriptionInput) throws DukeException {
+        if (descriptionInput[SECOND_ARRAY_PARAMETER].equals("") ||
+                descriptionInput[SECOND_ARRAY_PARAMETER].equals(" ")) {
+            throw new DukeException(ErrorMessage.EXCEPTION_MESSAGE_MISSING_PARAMETERS_AFTER_KEYWORD);
+        }
+    }
+
+    private static String[] descriptionInput(String[] words, String input) {
```

Comment:
Perhaps this method name should be a verb? https://se-education.org/guides/conventions/java/basic.html#naming